When I only open configurations `MEDIAINFO_TEXT_YES` and `MEDIAINFO_JSON_YES`, the compilation fails

> D:/dev/Mine/C_Project/MediaInfoLib/Source/MediaInfo/MediaInfo_Inform.cpp:832:21: error: unknown type name 'Text'; did you mean 'text'?
                if (Text)
                    ^~~~
                    text
D:/dev/Mine/C_Project/MediaInfoLib/Project/CMake/../../Source/MediaInfo/File__Analyse_Automatic.h:1200:6: note: 'text' declared here
enum text
     ^
D:/dev/Mine/C_Project/MediaInfoLib/Source/MediaInfo/MediaInfo_Inform.cpp:832:25: error: expected unqualified-id
                if (Text)
                        ^
D:/dev/Mine/C_Project/MediaInfoLib/Source/MediaInfo/MediaInfo_Inform.cpp:844:21: error: unknown type name 'Text'; did you mean 'text'?
                if (Text)
                    ^~~~
                    text
D:/dev/Mine/C_Project/MediaInfoLib/Project/CMake/../../Source/MediaInfo/File__Analyse_Automatic.h:1200:6: note: 'text' declared here
enum text
     ^
D:/dev/Mine/C_Project/MediaInfoLib/Source/MediaInfo/MediaInfo_Inform.cpp:844:25: error: expected unqualified-id
                if (Text)
                        ^
D:/dev/Mine/C_Project/MediaInfoLib/Source/MediaInfo/MediaInfo_Inform.cpp:958:21: error: unknown type name 'Text'; did you mean 'text'?
                if (Text)
                    ^~~~
                    text
D:/dev/Mine/C_Project/MediaInfoLib/Project/CMake/../../Source/MediaInfo/File__Analyse_Automatic.h:1200:6: note: 'text' declared here
enum text
     ^
D:/dev/Mine/C_Project/MediaInfoLib/Source/MediaInfo/MediaInfo_Inform.cpp:958:25: error: expected unqualified-id
                if (Text)
                        ^

In line 687 of file `MediaInfoLib/Source/MediaInfo_Inform.cpp`, there is no judgment on `MEDIAINFO_JSON_YES`. Maybe it should be changed to:
```cpp
#if defined(MEDIAINFO_TEXT_YES) && (defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES) || defined(MEDIAINFO_CSV_YES))
```

--- a/Source/MediaInfo/MediaInfo_Inform.cpp
+++ b/Source/MediaInfo/MediaInfo_Inform.cpp
@@ -50,9 +50,7 @@
 #include "MediaInfo/MediaInfo_Internal.h"
 #include "MediaInfo/File__Analyze.h"
 #include "ThirdParty/base64/base64.h"
-# if defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES)
 #include "MediaInfo/OutputHelpers.h"
-#endif //MEDIAINFO_XML_YES || MEDIAINFO_JSON_YES
 
 //---------------------------------------------------------------------------
 #include <ctime>
@@ -63,7 +61,6 @@ namespace MediaInfoLib
 {
 
 //---------------------------------------------------------------------------
-#if defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES)
 Ztring Xml_Name_Escape_0_7_78 (const Ztring &Name)
 {
     Ztring ToReturn(Name);
@@ -97,7 +94,6 @@ Ztring Xml_Name_Escape_0_7_78 (const Ztring &Name)
 
     return ToReturn;
 }
-#endif //defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES)
 
 //---------------------------------------------------------------------------
 std::string URL_Encoded_Encode(const std::string& URL);
@@ -636,7 +632,7 @@ Ztring MediaInfo_Internal::Inform()
 }
 
 //---------------------------------------------------------------------------
-#if defined(MEDIAINFO_TEXT_YES) || defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_CSV_YES) || defined(MEDIAINFO_CUSTOM_YES)
+#if defined(MEDIAINFO_TEXT_YES) || defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES) || defined(MEDIAINFO_CSV_YES) || defined(MEDIAINFO_CUSTOM_YES)
 #if defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES)
 namespace
 {
@@ -685,7 +681,7 @@ Ztring MediaInfo_Internal::Inform (stream_t StreamKind, size_t StreamPos, bool I
         #if defined(MEDIAINFO_CSV_YES)
         bool CSV=MediaInfoLib::Config.Inform_Get()==__T("CSV")?true:false;
         #endif //defined(MEDIAINFO_CSV_YES)
-        #if defined(MEDIAINFO_TEXT_YES) && (defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_CSV_YES))
+        #if defined(MEDIAINFO_TEXT_YES) && (defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES) || defined(MEDIAINFO_CSV_YES))
         bool Text=true;
         #if defined(MEDIAINFO_HTML_YES)
          if (HTML)
@@ -856,12 +852,13 @@ Ztring MediaInfo_Internal::Inform (stream_t StreamKind, size_t StreamPos, bool I
                 }
 
                 //Handling ISO-6709 location data for Text and HTML display
+                #if defined(MEDIAINFO_TEXT_YES) || defined(MEDIAINFO_HTML_YES)
                 bool TextOrHTML = false;
-                #if defined(MEDIAINFO_TEXT_YES) && (defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_CSV_YES))
+                #if defined(MEDIAINFO_TEXT_YES) && (defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES) || defined(MEDIAINFO_CSV_YES))
                 if (Text) TextOrHTML = true;
-                #else
+                #elif !(defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES) || defined(MEDIAINFO_CSV_YES))
                 TextOrHTML = true;
-                #endif // defined(MEDIAINFO_TEXT_YES) && (defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_CSV_YES))
+                #endif // defined(MEDIAINFO_TEXT_YES) && (defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES) || defined(MEDIAINFO_CSV_YES))
                 #if defined(MEDIAINFO_HTML_YES)
                 if (HTML) TextOrHTML = true;
                 #endif // defined(MEDIAINFO_HTML_YES)
@@ -872,6 +869,7 @@ Ztring MediaInfo_Internal::Inform (stream_t StreamKind, size_t StreamPos, bool I
                     if (std::regex_match(ISO6709, match, pattern))
                         Valeur.From_UTF8(match[1].str() + " " + match[2].str() + " " + match[3].str() + " " + match[4].str());
                 }
+                #endif // defined(MEDIAINFO_TEXT_YES) || defined(MEDIAINFO_HTML_YES)
 
                 #if defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES)
                     if ((XML_0_7_78 || JSON) && MediaInfoLib::Config.Info_Get(StreamKind).Read(Champ_Pos, Info_Measure)==__T(" ms"))

--- a/Source/MediaInfo/MediaInfo_Internal.h
+++ b/Source/MediaInfo/MediaInfo_Internal.h
@@ -79,7 +79,7 @@ public :
 
     //General information
     Ztring  Inform ();
-#if defined(MEDIAINFO_TEXT_YES) || defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_CSV_YES) || defined(MEDIAINFO_CUSTOM_YES)
+#if defined(MEDIAINFO_TEXT_YES) || defined(MEDIAINFO_HTML_YES) || defined(MEDIAINFO_XML_YES) || defined(MEDIAINFO_JSON_YES) || defined(MEDIAINFO_CSV_YES) || defined(MEDIAINFO_CUSTOM_YES)
     Ztring  Inform (stream_t StreamKind, size_t StreamNumber, bool IsDirect); //All about only a specific stream
 #endif
 

